cli/commands/site/stop-all.ts

```diff
@@ -0,0 +1,134 @@
+import { __, sprintf } from '@wordpress/i18n';
+import { SiteCommandLoggerAction as LoggerAction } from 'common/logger-actions';
+import { clearSiteLatestCliPid, readAppdata, type SiteData } from 'cli/lib/appdata';
+import { connect, disconnect } from 'cli/lib/pm2-manager';
+import { stopProxyIfNoSitesNeedIt } from 'cli/lib/site-utils';
+import { isServerRunning, stopWordPressServer } from 'cli/lib/wordpress-server-manager';
+import { Logger, LoggerError } from 'cli/logger';
+import { StudioArgv } from 'cli/types';
+
+const logger = new Logger< LoggerAction >();
+
+const filterRunningSites = async ( sites: SiteData[] ): Promise< SiteData[] > => {
+	const runningSites = [];
+
+	for ( const site of sites ) {
+		const runningProcess = await isServerRunning( site.id );
```

Comment:
[Here](https://github.com/Automattic/studio/pull/2207/files#diff-02b1fa12155a9878703b043a84451117eea52c0a5a20544cffec76272cc38a7fL97) we also check for `site.customDomain` and I am thinking - should we also do it here and for `studio site stop` command?